sdk/identity/identity/test/public/node/deviceCodeCredential.spec.ts

```diff
@@ -88,8 +94,11 @@ describe("DeviceCodeCredential", function() {
     } catch (e) {
       error = e;
     }
-    assert.equal(error?.name, "AbortError");
-    assert.equal(error?.message, "Cancellation triggered by the AbortSignal");
+
+    assert.equal(error?.name, "AuthenticationRequiredError");
```

Comment:
Someone else working on this repo would expect the error to be `AbortError` because that is what we associate with the abort signal. Are we getting `AuthenticationRequiredError` from MSAL or is this something that we throw?

Reply:
Good point. Let me take a better look at this.

Reply:
MSAL returns a `ClientAuthError`. I'll add a commit showing how we would make it an `AbortError`.

Reply:
I pushed an example of how we can capture and re-throw the error: https://github.com/Azure/azure-sdk-for-js/pull/14881/commits/dd74bb1611b8fae3984595626c82312af4228c84